📖 Improve ROSA quickstart with MachinePool and troubleshooting guide (#4775)
kubernetes-sigs/cluster-api-provider-aws: docs/book/src/topics/rosa/enabling.md

```shell
export EXP_ROSA="true"
export EXP_MACHINE_POOL="true"
clusterctl init --infrastructure aws
```

> ## TroubleShoot

Reviewer: ```suggestion
## Troubleshooting
```

> To check the feature-gates for the capi contoller Run the following command.

Reviewer: ```suggestion
To check the feature-gates for the Cluster API controller run the following command:
```

```shell
$ kubectl get deploy capi-controller-manager -n capi-system -o yaml
```

> check the feature gate container args it should has the MachinePool=true as below.

Reviewer 2: ```suggestion
the feature gate container arg it should have `MachinePool=true` as shown below.
```

Author: done

```yaml
spec:
      containers:
      - args:
        - --feature-gates=MachinePool=true,ClusterResourceSet=true,ClusterTopology=true,RuntimeSDK=false,MachineSetPreflightChecks=false
```

> To check the feature-gates for the capa contoller Run the following command.

Reviewer: ```suggestion
To check the feature gates for the Cluster API AWS controller run the following command:
```

```shell
$ kubectl get deploy capa-controller-manager -n capa-system -o yaml
```

> check the feature gate container args it should has the ROSA=true as below.

Reviewer 2: ```suggestion
the feature gate arg should have `ROSA=true` as shown below.
```

```yaml
spec:
    containers:
      - args:

>         - --feature-gates=ROSA=true,...

Reviewer: change to 2 spacing for better readability?

Author: done
